app/uk/gov/hmrc/nationalinsurancerecord/controllers/ErrorHandling.scala

```diff
@@ -33,7 +34,7 @@ trait ErrorHandling {
   def errorWrapper(func: => Future[Result])(implicit hc: HeaderCarrier): Future[Result] = {
     func.recover {
       case _: NotFoundException => NotFound(Json.toJson(ErrorNotFound))
-      case e: GatewayTimeoutException => Logger.error(s"$app Gateway Timeout: ${e.getMessage}", e); GatewayTimeout
+      case WithStatusCode(GATEWAY_TIMEOUT, e) => Logger.error(s"$app Gateway Timeout: ${e.getMessage}", e); GatewayTimeout
       case e: BadGatewayException => Logger.error(s"$app Bad Gateway: ${e.getMessage}", e); BadGateway
       case _: BadRequestException => BadRequest(Json.toJson(ErrorGenericBadRequest("Upstream Bad Request. Is this customer below State Pension Age?")))
       case e: Upstream4xxResponse => Logger.error(s"$app Upstream4XX: ${e.getMessage}", e); BadGateway
```

Comment:
Is leaving the deprecated Upstream4xxResponse/Upstream5xxResponse instead on UpstreamErrorResponse intentional?